var/spack/repos/builtin/packages/dpcpp/package.py

```diff
@@ -0,0 +1,121 @@
+# Copyright 2013-2021 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+
+import os
+
+from spack import *
+
+
+class Dpcpp(CMakePackage):
+    """Data Parallel C++ compiler: Intel's implementation of SYCL programming model"""
+
+    homepage = 'https://intel.github.io/llvm-docs/'
+    git = 'https://github.com/intel/llvm.git'
+
+    version('develop', branch='sycl')
+    version('stable', commit='7981e6027a8cb43028be55d19f12cb1e30a28f5a')
+
+    maintainers = ['ravil-mobile']
+    variant('cuda', default=False, description='switch from OpenCL to CUDA')
+    variant('rocm', default=False, description='switch from OpenCL to ROCm')
+    variant('rocm-platform', default='AMD', values=('AMD', 'NVIDIA'), multi=False, description='choose ROCm backend')
+    variant('arm', default=False, description='build ARM support rather than x86')
+    variant('disable-esimd-cpu', default=False, description='build without ESIMD_CPU support')
+    variant('no-assertions', default=False, description='build without assertions')
+    variant('docs', default=False, description='build Doxygen documentation')
+    variant('no-werror', default=False, description='Do not treat warnings as errors')
+    variant('shared-libs', default=False, description='Build shared libraries')
+    variant('use-lld', default=False, description='Use LLD linker for build')
```

Comment:
```suggestion
    variant('esimd_cpu', default=True, description='build with ESIMD_CPU support')
    variant('assertions', default=True, description='build with assertions')
    variant('docs', default=False, description='build Doxygen documentation')
    variant('shared', default=False, description='Build shared libraries')
    variant('lld', default=False, description='Use LLD linker for build')
```
I prefer to have variants with positive names. Otherwise you end up with double negatives like `~disable-...`, which makes a lot less sense than `+enable-...`. Also, things like `no-werror` do not affect what gets installed, only whether or not it can be built. For flags like these, it's better to add them unconditionally, or conditionally on the compiler. `+shared` is what we use in other packages. Let me know if you have any questions.

Reply:
Agree. I took these options directly from `intel/llvm` 

https://github.com/intel/llvm/blob/849774e32d01f9af8778524bed5656a2d72065d7/buildbot/configure.py#L159-L189

I will rework.

Btw, @adamjstewart, what do you think about the package name? Should it stay as `dpcpp` or it is better to rename to something `intel-dpcpp`?

Reply:
No preference on package name. Ask yourself if there are any other dpcpp libraries other than Intel's. Also ask yourself what people usually refer to the software as.

Reply:
> Ask yourself if there are any other dpcpp libraries other than Intel's

You are right. The official name is dpcpp.

> Also ask yourself what people usually refer to the software as.

Reply:
Hi @adamjstewart 
I also renamed variants as you suggested. Moreover, I added  
```
        if self.compiler.name == 'gcc':
            gcc_prefix = ancestor(self.compiler.cc, 2)
            args.append(self.define('GCC_INSTALL_PREFIX', gcc_prefix))
```

This is supposed to resolve linking problems caused by `libstd++`. I tested it on my system with one of my projects